src/compiler/parse/utils/__test__.ts

```diff
@@ -298,6 +298,30 @@ describe('decode_character_references', () => {
 		assert.equal(value, input);
 	});
 
+	it('escape character in attribute 1', () => {
+		// doesn't decode '&rect' which isn't html entity
+		const input = '<div style="background-image: url(https://url.com/path?lib=test&height=902&width=1500&rect=type1);"></div>';
+		const expect = '<div style="background-image: url(https://url.com/path?lib=test&height=902&width=1500&rect=type1);"></div>';
+		const value = decode_character_references(input);
+		assert.equal(value, expect);
+	});
+
+	it('escape character in attribute 2', () => {
+		// '&rect;' decode to ▭  which is html entity
+		const input = '<div style="background-image: url(https://url.com/path?lib=test&height=902&width=1500&rect;=type1);"></div>';
+		const expect = '<div style="background-image: url(https://url.com/path?lib=test&height=902&width=1500▭=type1);"></div>';
+		const value = decode_character_references(input);
+		assert.equal(value, expect);
+	});
+
+	it('escape character in attribute 3', () => {
+		// '&amp' decode to & which is html entity
+		const input = '<div style="background-image: url(https://url.com/path?lib=test&height=902&width=1500&amp=type1);"></div>';
+		const expect = '<div style="background-image: url(https://url.com/path?lib=test&height=902&width=1500&=type1);"></div>';
```

Comment:
I don't think this is correct and the reason I asked for the tests. If you paste this into an HTML document the browser will make the following request (I recommend always using `example.com` as it's made for this exact purpose):

![image](https://user-images.githubusercontent.com/679144/205682106-7fb6348c-8d22-426a-b360-1f0ab047b130.png)

So inside attributes the semicolon appears to be mandatory. If you add the semicolon this happens:

![image](https://user-images.githubusercontent.com/679144/205682550-c22fc0bb-31c0-4a6f-9f98-71a721a47b09.png)

Reply:
I don't know enough about the Svelte internals to comment on that. But from what I understand the whole reason we need to do this is that Svelte uses `textContent` and [`setAttribute`](https://github.com/sveltejs/svelte/blob/cc52dc442be8cb3207005d13bf9b1d521b2b0afd/src/runtime/internal/dom.ts#L292-L295). So the HTML entities need to have been decoded already by the compiler. So if you don't decode the string for attributes it's still not correct. Because `setAttribute('data-test', '&amp;')` will result in the literal `&amp;` being the attribute. But `<div data-test="&amp;" />` is `&` in HTML. Parsing the serialized HTML is a different thing from working with the de-serialized DOM.

Reply:
Yes, so it is no correct with no decode all attributes.

I think need to find out what needs to be decoded and what doesn't.

Reply:
I don't know enough about the Svelte internals too, beacuse this is my first PR, haha.

I think you know enough better than me.👍

Reply:
Sounds good, let's wait for a maintainer to comment on everything

Reply:
Ok,I going to start researching what needs to be decoded and what doesn't attribute.

And cases other than tags and content.

Reply:
I'm going to ask baseballyama  to review tomorrow.

It's too late in Japan now, so I'm afraid to disturb him.